### ISSUE TITLE
Manifest parse failure

### ISSUE THREAD

**Issue by alex-taffe** [compnerd/swift-build#303]
On the latest build of swift (downloaded via the python script), with a full uninstall and reinstall, the following error is occurring:
```
C:\SourceCache\redacted\desktop_client\shared\redacted >swift build
C:\SourceCache\AutoML\ redacted\shared\redacted: error: manifest parse error(s):
<unknown>:0: error: compiled module was created by an older version of the compiler; rebuild 'Swift' and try again: C:\Library\Developer\Platforms\Windows.platform\Developer\SDKs\Windows.sdk\usr\lib\swift\windows\Swift.swiftmodule\x86_64-unknown-windows-msvc.swiftmodule

C:\SourceCache\redacted\desktop_client\shared\redacted >
```

**Comment by compnerd** [comment]
That is very confusing.  The VS2019 builds use the just built Swift.  It seems like the SDK and the toolchain are out of sync.  Could you provide the build number that you used?  (I am trying to get a newer snapshot built, so I'm not 100% certain if I have something stupid in the script or if there is an issue in the build or the installer packaging).  I guess this is a good reason to try to figure out how to embed the version information into the build itself.

**Comment by ataffeqeexo** [comment]
```
C:\SourceCache\AutoML\desktop_client\shared\automl_core>swift --version
compnerd.org Swift version 5.3-dev (LLVM 8d6dfcd39f3bc7d, Swift 410b88ef07a1d18)
Target: x86_64-unknown-windows-msvc
```

**Comment by compnerd** [comment]
Yeah, I really need to get the build number embedded into that.  Okay, for the mean time, the URL would be handy I guess :-(.

**Comment by ataffeqeexo** [comment]
How do I actually find that? I just used the downloader script and have since closed my command prompt. I'm a macOS guy, sorry haha. I ran the python script about 10 minutes before posting this bug if that helps. If necessary I can also just upload the install exe somewhere like Google drive.

**Comment by compnerd** [comment]
Just drop the `--download` flag and it will print the URL :)

**Comment by ataffeqeexo** [comment]
The problem is the installer has most likely since changed because the CI has run since then and I was using the `--latest-artifacts` flag. I uploaded the installer to [Google Drive](https://drive.google.com/file/d/1BZ6jSXV_VCqh5ztBkmW1q_5WmsEKGWiR/view?usp=sharing). If the artifact link from the latest flag is at all helpful, it's [this](https://artprodeus21.artifacts.visualstudio.com/A8fd008a0-56bc-482c-ba46-67f9425510be/3133d6ab-80a8-4996-ac4f-03df25cd3224/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2NvbXBuZXJkL3Byb2plY3RJZC8zMTMzZDZhYi04MGE4LTQ5OTYtYWM0Zi0wM2RmMjVjZDMyMjQvYnVpbGRJZC8zOTIyMC9hcnRpZmFjdE5hbWUvaW5zdGFsbGVyLmV4ZQ2/content?format=zip)

**Comment by ataffeqeexo** [comment]
So I just ran the latest installer (the one that I posted in that last comment) and running `swift build` doesn't crash with the manifest parse issue, but rather hangs. Should I file a separate issue for that?

**Comment by compnerd** [comment]
I think that there's something stale left or the support files are out of date.  Could you re-copy the support files and remove the .build directory and then run swift build again?

**Comment by ataffeqeexo** [comment]
Hmm yeah I did both of those and it's still hanging

**Comment by compnerd** [comment]
That is very interesting.  I think that we should split that off into a separate issue.  I don't think that this is really the right avenue for the s-p-m issue - I think that the bugs.swift.org is probably better.  In any case, I've heard of one other case where the builder hangs, but I've not been able to figure out where the race condition it (I suspect that the process creation is going wrong and we aren't catching the error).  It is unclear whether it is in s-p-m or llbuild.  Would you mind running procmon and catching the `CreateProcess` calls and seeing if any of them are failing?

**Comment by ataffeqeexo** [comment]
Here's the procmon logfile in the PML format. I'm not a Windows guy so I was probably filtering wrong, but there didn't seem to be any CreateProcess calls (I doubt that's true but which is why I'm guessing I did it wrong)
[Logfile.zip](https://github.com/compnerd/swift-build/files/5307010/Logfile.zip)

**Comment by compnerd** [comment]
I see a number of calls to swiftc being made (which is normal during the startup process).  If you run with `-v` you should see those invocations.  What is interesting is that a file is being written for the stdout capture to redirect stdout, which is odd as I don't expect that.  But, I don't really see why the process would hang.

**Comment by ataffeqeexo** [comment]
Running with `-v` gave me:
```
C:\SourceCache\AutoML\desktop_client\shared\automl_core>swift build -v
C:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin\swiftc.exe -print-target-info
```

**Comment by compnerd** [comment]
I wonder if there is something waiting on the `-print-target-info` somehow.  Being able to get a reproduction would be rather useful, it seems like there is something in the s-p-m side that is racing in some cases (e.g. with the other instance, running the swift build procmon actually caused the problem to disappear).

**Comment by ataffeqeexo** [comment]
Attached is an example project that I was able to have the same issue occur on. Tested on [this](https://artprodeus21.artifacts.visualstudio.com/A8fd008a0-56bc-482c-ba46-67f9425510be/3133d6ab-80a8-4996-ac4f-03df25cd3224/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2NvbXBuZXJkL3Byb2plY3RJZC8zMTMzZDZhYi04MGE4LTQ5OTYtYWM0Zi0wM2RmMjVjZDMyMjQvYnVpbGRJZC8zOTIyMC9hcnRpZmFjdE5hbWUvaW5zdGFsbGVyLmV4ZQ2/content?format=zip) build

[reproduction.zip](https://github.com/compnerd/swift-build/files/5308089/reproduction.zip)

**Comment by ataffeqeexo** [comment]
So running `swift build` with the release version posted to the blog gives me this:
```
C:\SourceCache\redacted\desktop_client\shared\redacted>swift build
error: unable to invoke subcommand: C:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin\swift-build ()
```

I did a full uninstall and reinstall. Additionally, with the release, the REPL just immediately terminates with no output

**Comment by compnerd** [comment]
That is 100% expected.  swift-package-manager is not supported in the 5.3 release, so that wont work.  It has to be from the snapshots.  The REPL issue on the release is a known thing, there was a missed patch.  Unfortunately, I found some follow up issues due to the layout with LLDB but those can be worked around.

**Comment by compnerd** [comment]
I tried the reproduction that you posted - please note that this is not going to work off the bat - it is using a number of projects which aren't fully ported or at all (e.g. vapor).  That said, it doesn't hang for me, I've attached a log of what I saw with `swift build`.
[log.txt](https://github.com/compnerd/swift-build/files/5314026/log.txt)

**Comment by compnerd** [comment]
Note that this is largely what I expected - the problem isnt the package but rather the environment itself - there is some race condition in spm or llbuild that is getting exposed, and just needs to be tracked down.  Unfortunately, that isn't really possible to diagnose without a trace of what exactly is going on.

**Comment by ataffeqeexo** [comment]
Ok so something was just actually ruined on this machine environment. It got to the point that the C compiler wasn't working, let alone swift. I just wiped windows, installed a new copy of VS, Swift, git, etc. There are still dependency  issues  with the included packages, but running `swift build -Xlinker /INCREMENTAL:NO -v` did at least work

**Comment by Guang1234567** [comment]
Just reference to the same problem i met, thanks.

https://github.com/compnerd/swift-win32/issues/327#issuecomment-747231939
